pkg/publish/default.go

```diff
@@ -217,6 +217,11 @@ func (d *defalt) Publish(ctx context.Context, br build.Result, s string) (name.R
 			return nil, err
 		}
 
+		err = remote.CheckPushPermission(tag, d.keychain, d.t)
```

Comment:
Since we don't have to check this permission every time, moving that part from the loop would be better. Also, there are different types of publishers, and this change only affects the default one, so it'd be better to move that logic into the common place where we call all the configured publishers.

PTAL @imjasonh